### PR TITLE
Add DMS MigrationJob resource 

### PR DESCRIPTION
**Pull request by owaismawasy**
Add the DMS resource `MigrationJob` to the Terraform Google provider. 

```release-note:new-resource
`google_database_migration_service_migration_job`
```

### CODE REVIEW

File: mmv1/products/databasemigrationservice/MigrationJob.yaml [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    name: 'staticIpConnectivity'
+    description: |
+      static ip connectivity data (default, no additional details needed).
+    exactly_one_of:
+      - reverseSshConnectivity
+      - vpcPeeringConnectivity
+    allow_empty_object: true
+    send_empty_value: true
+    properties: []
+  - !ruby/object:Api::Type::NestedObject
```
Comment (rileykarson) [inline mmv1/products/databasemigrationservice/MigrationJob.yaml:238]: This object+below should any of the subfields be required?

File: mmv1/products/databasemigrationservice/MigrationJob.yaml [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    description: |
+      The type of the data dump. Supported for MySQL to CloudSQL for MySQL
+      migrations only.
+    values:
+      - :LOGICAL
+      - :PHYSICAL
+  - !ruby/object:Api::Type::NestedObject
+    name: 'staticIpConnectivity'
+    description: |
+      static ip connectivity data (default, no additional details needed).
```
Comment (rileykarson) [inline mmv1/products/databasemigrationservice/MigrationJob.yaml:231]:
```suggestion
     If set to an empty object (`{}`), the source database will allow incoming connections from the public IP of the destination database. You can retrieve the public IP of the Cloud SQL instance from the Cloud SQL console or using Cloud SQL APIs.
```

Text from https://cloud.google.com/database-migration/docs/reference/rest/v1/StaticIpConnectivity

File: mmv1/templates/terraform/examples/database_migration_service_migration_job_postgres_to_postgres.tf.erb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+resource "google_database_migration_service_migration_job" "<%= ctx[:primary_resource_id] %>" {
+  location              = "us-central1"
+  migration_job_id = "<%= ctx[:vars]["migration_id"] %>"
+  display_name = "<%= ctx[:vars]["migration_id"] %>_display"
+  labels = {
+    foo = "bar"
+  }
+  static_ip_connectivity {
+  }
+  source             = "projects/${data.google_project.project.project_id}/locations/us-central1/connectionProfiles/<%= ctx[:vars]["source_cp"] %>"
```
Comment (rileykarson) [inline mmv1/templates/terraform/examples/database_migration_service_migration_job_postgres_to_postgres.tf.erb:84]: Here + other connection profile URLs, can we use a reference to their `id` field? That would also allow removing the `depends_on`.

File: mmv1/products/databasemigrationservice/MigrationJob.yaml [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+        description: |
+          Human readable message indicating details about the current status.
+      - !ruby/object:Api::Type::Array
+        name: 'details'
+        description: |
+          A list of messages that carry the error details.
+        item_type: Api::Type::KeyValuePairs
+        output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'type'
```
[the remark this answers is not in the document]
Comment (owaismawasy) [reply mmv1/products/databasemigrationservice/MigrationJob.yaml:168]: This one is immutable. I marked immutable fields as such, and will add an update test for the other ones.

File: mmv1/products/databasemigrationservice/MigrationJob.yaml [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+  - !ruby/object:Api::Type::NestedObject
+    name: 'dumpFlags'
+    description: |
+      The initial dump flags.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: 'dumpFlags'
+        description: |
+          A list of dump flags
+        item_type: !ruby/object:Api::Type::NestedObject
```
[the remark this answers is not in the document]
Comment (owaismawasy) [reply mmv1/products/databasemigrationservice/MigrationJob.yaml:191]: No, these are optional.

File: mmv1/products/databasemigrationservice/MigrationJob.yaml [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+          - :OPTIMAL
+          - :MAX
+  - !ruby/object:Api::Type::String
+    name: 'dumpPath'
+    description: |
+      The path to the dump file in Google Cloud Storage,
+      in the format: (gs://[BUCKET_NAME]/[OBJECT_NAME]).
+      This field and the "dump_flags" field are mutually exclusive.
+  - !ruby/object:Api::Type::Enum
+    name: 'dumpType'
```
[the remark this answers is not in the document]
Comment (owaismawasy) [reply mmv1/products/databasemigrationservice/MigrationJob.yaml:221]: This is optional, and in this case the API will use a default value behind the scene but it won't change the value.

File: mmv1/products/databasemigrationservice/MigrationJob.yaml [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+          for the SSH tunnel.
+      - !ruby/object:Api::Type::Integer
+        name: 'vmPort'
+        description: |
+          The forwarding port of the virtual machine (Compute Engine) used as the
+          bastion server for the SSH tunnel.
+      - !ruby/object:Api::Type::String
+        name: 'vm'
+        description: |
+          he name of the virtual machine (Compute Engine) used as the bastion server
```
[the remark this answers is not in the document]
Comment (owaismawasy) [reply mmv1/products/databasemigrationservice/MigrationJob.yaml:259]: Done.

File: mmv1/products/databasemigrationservice/MigrationJob.yaml (hunk 1 above)
Comment (owaismawasy) [reply mmv1/products/databasemigrationservice/MigrationJob.yaml:238]: Marked required fields as such.

File: mmv1/products/databasemigrationservice/MigrationJob.yaml (hunk 2 above)
Comment (owaismawasy) [reply mmv1/products/databasemigrationservice/MigrationJob.yaml:231]: Done.

File: mmv1/templates/terraform/examples/database_migration_service_migration_job_postgres_to_postgres.tf.erb (hunk 3 above)
Comment (owaismawasy) [reply mmv1/templates/terraform/examples/database_migration_service_migration_job_postgres_to_postgres.tf.erb:84]: Done.

**Review by rileykarson — APPROVED** [review]

**Comment by rileykarson** [comment]
LGTM, thanks!

**Pull request merged**